config/configgrpc/configgrpc.go

```diff
@@ -190,7 +168,7 @@ func (gcs *ClientConfig) ToClientConn(ctx context.Context, host component.Host,
 		return nil, err
 	}
 	opts = append(opts, extraOpts...)
-	return grpc.DialContext(ctx, gcs.SanitizedEndpoint(), opts...)
+	return grpc.DialContext(ctx, gcs.Endpoint, opts...)
```

Comment:
This feels like a breaking change we shouldn't be doing.  Won't this affect users of the `otlpreceiver` how do something like:

```yaml
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: https://my-address:4317
```

If the intent was only to hide this function publicly I can add this feature back privately.

Reply:
Very confused about this comment. Please explain in the PR description

Reply:
@bogdandrutu I've removed all `SanitizedEndpoint` deprecations from this PR until we can discuss https://github.com/open-telemetry/opentelemetry-collector/issues/9482 more so you can disregard my comment in the context of this PR.